app/journey/RemovalReasonType.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package journey
+
+import play.api.libs.json.{Format, JsError, JsResult, JsString, JsSuccess, JsValue}
+
+sealed trait RemovalReasonType
+
+case object Demolition extends RemovalReasonType
+case object Disrepair extends RemovalReasonType
+case object Renovating extends RemovalReasonType
+case object NotComplete extends RemovalReasonType
+case object BandedTooSoon extends RemovalReasonType
+case object CaravanRemoved extends RemovalReasonType
+case object Duplicate extends RemovalReasonType
+case object OtherReason extends RemovalReasonType
+
+
+object RemovalReasonType {
+  val order = List[String] (
+    "Demolition",
+    "Disrepair",
+    "Renovating",
+    "NotComplete",
+    "BandedTooSoon",
+    "CaravanRemoved",
+    "Duplicate",
+    "OtherReason"
+  )
+
+  implicit val format: Format[RemovalReasonType] = new Format[RemovalReasonType] {
+    override def reads(json: JsValue): JsResult[RemovalReasonType] = {
+      json match  {
+        case JsString("It has been demolished") => JsSuccess(Demolition)
+        case JsString("It is in disrepair") => JsSuccess(Disrepair)
```

Comment:
Shouldn't be `JsString("RemoveProperty")` Same as in `writes` ?

Reply:
🤦 Corrected in https://github.com/hmrc/voa-bar-xml-frontend/pull/201/commits/fcedcf04a33aaad20697db8a502c34cee7651100

Reply:
Why we serialize to Json as `It has been demolished` instead of `Demolition`

Reply:
None I guess https://github.com/hmrc/voa-bar-xml-frontend/pull/201/commits/12399e9a3a985cf9fdce8cd738ef5c423be1b045